src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,70 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename, bool binary)
+    : filename_(filename)
+{
+  std::string tmp_filename = filename_ + ".tmp";
+  if (fs::exists(tmp_filename)) {
+    fs::remove(tmp_filename);
+  }
+  os_.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+  try {
+    if (binary) {
+      os_.open(tmp_filename, std::ios::binary | std::ios_base::out);
```

Comment:
Nit: I think the following reads a bit better than an if-statement.

os_.open(tmp_filename, (binary ? std::ios::binary : 0) | std::ios_base::out);